starknet_py/net/full_node_client.py

```diff
@@ -385,24 +385,32 @@ async def estimate_message_fee(
         block_identifier = get_block_identifier(
             block_hash=block_hash, block_number=block_number
         )
-        assert Web3.is_address(
+
+        assert is_valid_eth_address(
             from_address
         ), f"Argument 'from_address': {from_address} is not a valid Ethereum address."
+
         message_body = {
             "from_address": from_address,
             "to_address": _to_rpc_felt(to_address),
             "entry_point_selector": _to_rpc_felt(entry_point_selector),
             "payload": [_to_rpc_felt(x) for x in payload],
         }
 
-        res = await self._client.call(
-            method_name="estimateMessageFee",
-            params={
-                "message": message_body,
-                **block_identifier,
-            },
-        )
-        return cast(EstimatedFee, EstimatedFeeSchema().load(res, unknown=EXCLUDE))
+        try:
+            res = await self._client.call(
+                method_name="estimateMessageFee",
+                params={
+                    "message": message_body,
+                    **block_identifier,
+                },
+            )
+            return cast(EstimatedFee, EstimatedFeeSchema().load(res, unknown=EXCLUDE))
+        except ClientError as err:
+            raise ClientError(
```

Comment:
Maybe we could specifically check if the RPC error is the Contract Error or whatever is returned when the L1 address is invalid